src/into_dart.rs

```diff
@@ -184,14 +184,18 @@ where
     assert_eq!(length, vec.capacity());
     let ptr = vec.as_mut_ptr();
 
+    let raw_slice = Box::into_raw(Box::new(
+        std::ptr::slice_from_raw_parts_mut(ptr, length),
+    ));
```

Comment:
nit: `slice` has the semantics of (something similar to) "a borrowed vec", i.e. some lifetime things, thus maybe confuse the reader a little bit. Thus is the following alternative possible:

```rust
struct VecInfo { ptr, len }
let raw_info = Box::into_raw(Box::new(VecInfo { ptr, length }));
```

and recover it

```rust
let raw_info = Box::from_raw(peer);
Vec::from_raw_parts(raw_info.ptr, raw_info.len, raw_info.len)
```

Reply:
This is so called raw slice, it is `*mut [u8]` has no lifetime. I just needed a structure to deliver both ptr and length, I could have created one, but I used what was already there.

Reply:
You are right!

Reply:
@fzyzcjy Please take a look again. I just figure out the simplest solution.